### ISSUE TITLE
Onlyoffice working terrible bad/not working

### ISSUE THREAD

**Issue by PolGZ** [hyprwm/Hyprland#1874]
I only have hyperland installed right now, so I can't check if the problem persists on full DE's or other WM's. Sorry if it's not a hyprland-related issue.

I have installed Onlyoffice **for the first time** (so I don't know if it happened on previous versions), to see if it would behave better with a few .doc documents from a colleague and it preserves the structure of the document better. 

I've tried both flatpak (v. 7.3.0) and AUR [onlyoffice-bin](https://aur.archlinux.org/packages/onlyoffice-bin) package (v. 7.3.3-1). Both of them are **terribly** slow, to the point that it is completely unusable. Maybe half a minute to open and be able to make a single click. 10 seconds to open a file, after 20 seconds to click the open menu... 20 seconds to scroll or type... same to adjust the size if you change the window size...

hyprland version: hyprland-bin 0.23.0beta-1

If I launch it from the terminal (the AUR version), I get these errors:

```
QXcbConnection: XCB error: 8 (BadMatch), sequence: 29556, resource id: 35651776, major code: 130 (Unknown), minor code: 3
QXcbConnection: XCB error: 8 (BadMatch), sequence: 29561, resource id: 35651776, major code: 130 (Unknown), minor code: 3
QXcbConnection: XCB error: 8 (BadMatch), sequence: 29564, resource id: 35651776, major code: 130 (Unknown), minor code: 3
QXcbConnection: XCB error: 8 (BadMatch), sequence: 29588, resource id: 35651776, major code: 130 (Unknown), minor code: 3
```

**Comment by vaxerski** [comment]
does the same happen on sway?

**Comment by Desktop31** [comment]
Hi, today I noticed I have the same issue with Onlyoffice, however, it seems to affect more applications

TIMELINE:
---

**DESKTOP (RX6700XT)**
11.3.
  Hyprland cmake build (Feb 19): GTA V (lutris, wine) no issues

23.3.
  Hyprland cmake build (Feb 19): GTA V extremely laggy (Rockstar Launcher works ok)

25.3.
  Hyprland cmake build (Feb 19): Onlyoffice extremely laggy

  *Updated Hyprland*
  Hyprland cmake build (March 25): Onlyoffice still laggy
  
**LAPTOP (Iris Xe Graphics)**
25.3.
  Hyprland cmake build (Feb 20): Onlyoffice extremely laggy

  DWM (Xorg): Onlyoffice no issues
  
  Gnome Wayland: Onlyoffice no issues

  Gnome Xorg: Onlyoffice no issues

---

**Additional info:**
Onlyoffice is using xwayland
GDLauncher and Minecraft 1.19 both work without any issues under xwayland

**Observation:**
  When I quickly switch from another window (e.g. kitty) to Onlyoffice, it works ok for a brief moment. 
  (The window does not seem to be getting updates when focused)

**TLDR:**
  Something GTA V and Onlyoffice have in common stopped working correctly between March 11th and 23rd, but this doesn't affect all xwayland applications nor Gnome Wayland.

**Video:** https://youtu.be/H0yJuddoAbA
(top of hyprland log in the description)

---
List of desktop packages: https://pastebin.com/DvXzJnKd
List of laptop packages: https://pastebin.com/phtgwwKB

[Desktop updates]
Pacman log on March 11th: https://pastebin.com/xLV6c9jq
Pacman log since March 11th: https://pastebin.com/bh6nb7ad

I can provide full log (I reinstalled linux on 31.1.2023). 
I also kept the Hyprland source code from Feb 19th install, so I can test more stuff if needed.

**Comment by vaxerski** [comment]
gnome wayland is the worst thing to compare to. Please compare against sway-git if anything.

> 11.3.
Hyprland cmake build (Feb 19): GTA V (lutris, wine) no issues

> 23.3.
Hyprland cmake build (Feb 19): GTA V extremely laggy (Rockstar Launcher works ok)

> 25.3.
Hyprland cmake build (Feb 19): Onlyoffice extremely laggy

Doesn't this suggest the issue is _not_ Hyprland but something else?

**Comment by Desktop31** [comment]
I tried sway and it doesn't work either. I looked it up and there is an issue open on the Onlyoffice github page, where it seems that versions prior to 7.3.0 do work.
Also wine was updated 21.3. and that might be a reason why GTA V doesn't work (won't even turn on rn so I can't test it), so I guess it is just 2 broken updates at once and the issue has likely nothing to do with Hyprland in the end.

**Comment by PolGZ** [comment]
I did a downgrade of the flatpak version to the previous commit (as shown in [flatpak wiki](https://github.com/flatpak/flatpak/wiki/Tips-&-Tricks) > "Downgrading") that shows an older release version (v. 7.2.1) and I can confirm it works fine.

I've never used onlyoffice before neither I knew I could downgrade flatpaks to previous versions... so sorry for not testing that before opening the issue 🙇.

**Comment by Froggo8311** [comment]
Onlyoffice is really bad currently, basically broken through Xwayland. There were earlier versions of it, and I'm not sure what they changed, but after an update it started being very sluggish. I created an issue on it- let me grab it. Not an issue with Hyprland, though.

**Comment by Froggo8311** [comment]
https://github.com/ONLYOFFICE/DesktopEditors/issues/1220
